source/keymanweb.js

```diff
@@ -3542,16 +3574,97 @@
+       * 
+       * We place it here so that it loads after most of the other UI loads, reducing the MutationObserver's overhead.
+       */
+
+      var observationTarget = document.querySelector('body');
+      keymanweb.mutationObserver = new MutationObserver(function(mutations) 
+        {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(var i=0; i < mutations.length; i++)
+          {
+            mutation = mutations[i];
+            
+            for(var j = 0; j < mutation.addedNodes.length; j++)
+            {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = "";
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) 
+              {
+                childAdditions = childAdditions.concat(
+                  addedNode.getElementsByTagName('input'),
+                  addedNode.getElementsByTagName('textarea'),
+                  addedNode.getElementsByTagName('iframe'),
+                );
+              }
+
+              if(lcTagName == 'input' || lcTagName == 'textarea' || lcTagName == 'iframe')
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(addedNode);
+              }
+
+              for(k = 0; k < childAdditions.length; k++)
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(childAdditions[k]);
+              }
+            }
+
+            // There also exists a 'mutation.removedNodes' array.  We'll need to address that for issue #63.
+
+            // After all mutations have been handled, we need to recompile our .sortedInputs array.
+            if(dirtyFlag) {
+              keymanweb.listInputs();
+            }
+          }
+        });
+
+      var observationConfig = { childList: true, subtree: true };
+      keymanweb.mutationObserver.observe(observationTarget, observationConfig);
+
     // Set exposed initialization flag to 2 to indicate deferred initialization also complete
     keymanweb['initialized']=2;
-  }  
+  }
+
+  // Used by the MutationObserver event handler to properly setup any elements dynamically added to the document post-initialization.
+  keymanweb._MutationAdditionObserved = function(Pelem)
+  {
+    if(!device.touchable)
+    {
+      // keymanweb.attachToControl is written to handle iframes, but setupDesktopElement is not.
+      if(keymanweb.setupDesktopElement(Pelem)) {
+        keymanweb.attachToControl(Pelem);
+      } else if(Pelem.tagName.toLowerCase() == 'iframe') {
+        //Problem:  the iframe is loaded asynchronously, and we must wait for it to load fully before hooking in.
```

Comment:
Admittedly, I think there is a small possibility that it could load by that time... but it's unclear how we could possibly intercept it without putting stringent requirements on our end users.

The one way to ensure that it can't load by this point is to have never added it in the first place.  Please see: http://stackoverflow.com/questions/3142837/capture-iframe-load-complete-event

We'd have to trap an iframe's addition to the document, block and capture it, and manually reinsert it without the page being affected in order to have a 100% guarantee, based on my understanding.